File: desktop/src/main/kotlin/host/flux/desktop/services/ProjectNameNormalizer.kt
package host.flux.desktop.services

object ProjectNameNormalizer {
    fun normalize(name: String): String {
        return name
            .lowercase()
            .replace(' ', '-')
            .replace(Regex("[^a-z0-9_-]"), "")
            .replace(Regex("[-_]{2,}"), "-")
            .trim('-', '_')
            .take(50)
            .trimEnd('-', '_')
    }
}

File: native/windows/FluxzeroLaunchpad.Core/ProjectNameNormalizer.cs
using System.Text.RegularExpressions;

namespace Fluxzero.Launchpad;

public static class ProjectNameNormalizer
{
    public static string Normalize(string name)
    {
        var cleaned = Regex.Replace(name.ToLowerInvariant(), @"[^a-z0-9\s_-]", "");
        var dashed = Regex.Replace(cleaned, @"[\s_]+", "-");
        return Regex.Replace(dashed, "-+", "-").Trim('-');
    }

    public static string PackageSuffix(string artifact)
    {
        var suffix = Regex.Replace(artifact.ToLowerInvariant(), @"[^a-z0-9]", "");
        return string.IsNullOrWhiteSpace(suffix) ? "app" : suffix;
    }
}
